string name
string updater
---
bool added
